### PR TITLE
add swap registry to migration script

### PR DESCRIPTION
**Pull request by johannbarbie**
resolves https://github.com/leapdao/leap-contracts/issues/128
resolves https://github.com/leapdao/leap-contracts/issues/130

- [x] deploy swap registry
- [x] allow to pass token as environment variable
- [x] add more environment variables
- [x] add all variables to readme
- [x] make token governable
- [x] reduce storage footprint of period

### CODE REVIEW

File: migrations/2_deploy_token.js [hunk 1]
```diff
@@ -10,8 +10,13 @@ const log = require('./utils/log');
 const NativeToken = artifacts.require('NativeToken');
 
 module.exports = (deployer) => {
+
+  const deployedToken = process.env.DEPLOYED_TOKEN;
+
   deployer.then(async () => {
-    const nativeToken = await deployer.deploy(NativeToken, "LeapToken", "LEAP", 18);
-    log('Deployed LEAP Token at', nativeToken.address);
+  	if (!deployedToken) {
```
Comment (troggy) [inline migrations/2_deploy_token.js]: still not sure this is the right way to use truffle migrations

File: migrations/4_deploy_governance.js [hunk 2]
```diff
@@ -35,5 +45,27 @@ module.exports = (deployer, network, accounts) => {
...
+    }
+
+    if (ownerAddr) {
+      log('  🔄 Transferring ownership for Governance:', ownerAddr);
+      await governance.transferOwnership(ownerAddr);
+    }
+
+    const isRegistryMinter = await nativeToken.isMinter(registryProxy.address);
+    if (!isRegistryMinter) {
+      log('  ⚠ Minting rights could not be set on token:', nativeToken.address);
```
Comment (troggy) [inline migrations/4_deploy_governance.js]: ❤️

File: migrations/4_deploy_governance.js [hunk 3]
```diff
@@ -17,9 +19,17 @@ const DEFAULT_PROPOSAL_TIME = duration.days(14);
...
 module.exports = (deployer, network, accounts) => {
   const admin = accounts[1];
+  const proposalTime = process.env.PROPOSAL_TIME || DEFAULT_PROPOSAL_TIME;
+  const ownerAddr = process.env.GOV_OWNER;
+  const deployedToken = process.env.DEPLOYED_TOKEN;
 
   deployer.then(async () => {
-    const proposalTime = process.env.PROPOSAL_TIME || DEFAULT_PROPOSAL_TIME;
+    let nativeToken;
+    if(deployedToken) {
```
Comment (troggy) [inline migrations/4_deploy_governance.js]: code duplication 🤓

File: migrations/4_deploy_governance.js [hunk 4]
```diff
@@ -35,5 +45,27 @@ module.exports = (deployer, network, accounts) => {
...
     await exitHandlerProxy.changeAdmin(governance.address, { from: admin });
+
+    const registryProxy = await AdminableProxy.deployed();
+    log('  🔄 Transferring ownership for SwapRegistry:', registryProxy.address);
+    await registryProxy.changeAdmin(governance.address, { from: admin });
+
+    const isMinter = await nativeToken.isMinter(accounts[0]);
+    if (isMinter) {
+      log('  🔄 Transferring minting right for token:', nativeToken.address);
+      await nativeToken.addMinter(governance.address);
```
Comment (troggy) [inline migrations/4_deploy_governance.js]: ❤️

File: migrations/2_deploy_token.js (hunk 1 above)
Comment (johannbarbie) [reply migrations/2_deploy_token.js]: agree, but reading through migration contract i didn't get any wiser 🤷‍♂️

File: test/minGov.js [hunk 5]
```diff
@@ -210,6 +211,20 @@ contract('MinGov', (accounts) => {
...
+  it('should allow to mint token', async () => {
+    const token = await NativeToken.new("Token", "TOK", 18);
+    await token.addMinter(gov.address);
+    // propose and finalize value change
+    const data = await token.contract.methods.addMinter(accounts[1]).encodeABI();
+    await gov.propose(token.address, data);
+    await gov.finalize();
+
+    // test minting rights
+    await token.mint(accounts[1], 100000);
```
Comment (troggy) [inline test/minGov.js]:
```suggestion
    await token.mint(accounts[1], 100000, { from: accounts[1] });
```

Comment (troggy) [reply test/minGov.js]: otherwise we are not testing minting rights

Comment (johannbarbie) [reply test/minGov.js]: good point

**Pull request merged**